adam_preprocessing/predict.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+#SBATCH --job-name=objectsGNN_predict
+#SBATCH --account=borrowed
+#SBATCH --partition=ephemeral
+#SBATCH --qos=ephemeral
+#SBATCH --time=4:00:00 # Number of hours required per node, max 24 on SAGA
+#SBATCH --ntasks=1
+#SBATCH --cpus-per-task=16
+#SBATCH --mem=32g
+#SBATCH --gpus-per-task=1
+#SBATCH --nodes=1
+#SBATCH --mail-type=FAIL,END
+#SBATCH --output=R-%x.%j.out
+# Use ephemeral because I don't think we have any GPUs on the adam partition.
+# Only request 1 GPU because we can't use any more -- the script does not use DP.
+set -u
+
+if [[ "$#" -ne 3 ]] || [[ "$1" = "--help" ]] ; then
+  printf '%s\n' "usage: $0 model_path input_curriculum_dir output_curriculum_dir"
+  python shape_stroke_graph_inference.py --help
+  exit 1
+else
+  model_path=$1
+  input_curriculum_dir=$2
+  output_curriculum_dir=$3
+  shift 3
+fi
+
+python shape_stroke_graph_inference.py "$model_path" "$input_curriculum_dir" --save_outputs_to "$output_curriculum_dir" "$@"
```

Comment:
Out of curiosity, after shifting over by 3, what is the `"$@"` used for here? Earlier, you check to make sure there are exactly 3 inputs to the script (`if [[ "$#" -ne 3 ]]`), so even if there are further arguments provided, the script would just terminate instead of adding the extra arguments to the end. It also looks like in `shape_stroke_graph_inference.py`, there are only 3 specified command line arguments anyway, so it makes sense to me that you only check for 3 in the script, just that the `"$@` wouldn't really do anything as is.

Reply:
The `"$@"` is as you thought not actually doing anything here. I think it came over out of habit or because I copied the setup from a Slurm script for another project where I use `-lt`.

I think it's good to have it there as a reminder in case we decide to add more args. But you're right, it's a confusing thing to include. I've added comments make this explicit.